### PR TITLE
Improve "Item route" related files

### PR DESCRIPTION
**Pull request by smadbe**
Item route has been built by several unrelated updates and has become quite messy and inconsistent.
In order to prepare for the introduction of route for groups very soon, I tried to clean item route(r) and make utility function more consistent.

It is recommended to do this review commit per commit.

### CODE REVIEW

File: src/app/shared/helpers/url.ts [hunk 1]
```diff
@@ -55,3 +56,20 @@ export function clearHash(paramNames: string[]): void {
...
+export function urlStringFromArray(urlAsArray: (string|{[k: string]: string|string[]})[]): string {
+  // do not try to understand the implementation, just check the tests
+  return urlAsArray.map((part, idx) => {
+    if (part === '/') return '';
+    else if (isString(part)) return (idx === 0 ? '' : '/') + part;
+    else return Object.keys(part).map(key => {
+      const val = part[key];
+      if (val === undefined) throw new Error('unexpected: cannot find key in a dict while iterating over keys');
+      const valStr = isString(val) ? val : val.join(',');
+      return ';' + key + '=' + valStr;
```
Comment (Iloveall) [inline src/app/shared/helpers/url.ts:75]: More clear for me:

`;${ key }=${ valStr }`;

File: src/app/shared/routing/item-route.ts [hunk 2]
```diff
@@ -41,30 +44,12 @@ export function appDefaultItemRoute(cat: ItemTypeCategory = 'activity'): ItemRou
...
-function urlArrayForItem(id: ItemId, cat: ItemTypeCategory, params: UrlCommandParameters, page: string|string[]):
-  UrlCommand {
-  return [ '/', itemTypeCategoryPrefix(cat), 'by-id', id, params, ...pageToPagePath(page) ];
-}
+/* **********************************************************************************************************
+ * Utility functions for finding the item route from router information
+ * ********************************************************************************************************** */
 
-interface ItemRouteError {
+ interface ItemRouteError {
```
Comment (Iloveall) [inline src/app/shared/routing/item-route.ts:52]: Unexpected Indent

**Review by Iloveall — CHANGES_REQUESTED** [review]
I have left small remarks

File: src/app/shared/helpers/url.ts (hunk 1 above)
Comment (smadbe) [reply src/app/shared/helpers/url.ts]: done

File: src/app/shared/routing/item-route.ts (hunk 2 above)
Comment (smadbe) [reply src/app/shared/routing/item-route.ts]: fixed 👍

**Review by Iloveall — APPROVED** [review]

**Pull request merged**